### PR TITLE
build: Move to webpack 4

### PR DESCRIPTION
**Pull request by martinpitt**
This is a prerequisite for moving to babel 7, and blocks the update of a lot of other npm modules.

 - [x] Build system fixes #10834

### CODE REVIEW

**Comment by martinpitt** [comment]
In a production build, dist/ shrinks from 187 MB to 110 MB with this. \o/

**Comment by martinpitt** [comment]
This breaks some openshift test, investigating.

**Comment by martinpitt** [comment]
The [testVolumes](https://fedorapeople.org/groups/cockpit/logs/pull-10826-20181219-065959-195680d0-verify-fedora-29/log.html#16) failure is actually a race condition which has been there all the time. It is just more apparent now due to slightly different timing  behaviour.

The [test](https://github.com/cockpit-project/cockpit/blob/master/test/verify/kubelib.py#L311) tries to assert a Persistent Volume Claim in state "bound", which is wrong: It's a claim while it's in state "pending", and becomes  a Persistent Volume once bound. Thus the page will only show the PVC for a very short time until it turns into a PV.

This can be seen in the [screenshot](https://fedorapeople.org/groups/cockpit/logs/pull-10826-20181219-065959-195680d0-verify-fedora-29/TestKubernetes-testVolumes-fedora-29-127.0.0.2-2401-FAIL.png).

I verified interactively that current master exhibits the same behaviour -- when you open the page, it shows "Type: Persistent Volume".

I added a fix here. It could be a separate PR if you prefer, but with being on holidays I don't sit on the computer often enough to orchestrate them, so I hope it's ok (but do let me know, I can split it out,  too).

File: Makefile.am [hunk 1]
```diff
@@ -205,10 +205,11 @@ dist/%/Makefile.deps:
 	  cp -p $(srcdir)/$@ $@.tmp && \
 	  mv $@.tmp $@)
 
-# This is the primary rule for running webpack
+# This is the primary rule for running webpack; some pkgs like ssh or pcp only
+# have a manifest, no actual webpack, so don't call webpack-make for these
 dist/%/stamp: $(WEBPACK_CONFIG) $(srcdir)/tools/webpack-make
 	$(V_WEBPACK) $(MKDIR_P) dist/$* && \
-	  $(WEBPACK_MAKE) -d dist/$*/Makefile.deps $(WEBPACK_CONFIG) && \
+	(if ls $(srcdir)/pkg/$*/*.html >/dev/null 2>&1; then $(WEBPACK_MAKE) -d dist/$*/Makefile.deps $(WEBPACK_CONFIG); else touch dist/$*/Makefile.deps; fi ) && \
```
Comment (mvollmer) [inline Makefile.am]: What about moving this logic into webpack-make itself?

Comment (martinpitt) [reply Makefile.am]: I thought about it, but I was a bit concerned about these things:

 * it's a bit confusing to call "webpack-make" on something that isn't making a webpack at all
 * starting a full nodejs interpreter to essentially do nothing is a waste
 * globbing is a bit more involved in JS

But I'm not opposed to changing this. (Then again, our build system is horrificly complicated either way.. :cry: )

**Pull request merged**